packages/plugins/apollo-managed-federation/src/index.ts

```diff
@@ -1,15 +1,66 @@
 import type { Plugin } from 'graphql-yoga';
-import { SupergraphSchemaManager, SupergraphSchemaManagerOptions } from '@graphql-tools/federation';
+import { createGraphQLError } from 'graphql-yoga';
+import {
+  FetchError,
+  SupergraphSchemaManager,
+  SupergraphSchemaManagerOptions,
+} from '@graphql-tools/federation';
 
-export type ManagedFederationPluginOptions = SupergraphSchemaManagerOptions;
-
-export type Logger = {
-  info: (message: string, ...args: unknown[]) => void;
-  error: (message: string, error?: unknown, ...args: unknown[]) => void;
+export type ManagedFederationPluginOptions = (
+  | (SupergraphSchemaManagerOptions & { supergraphManager?: never })
+  | { supergraphManager: SupergraphSchemaManager }
+) & {
+  /**
+   * The manager to be used for fetching the schema and keeping it up to date
+   * If not provided, on will be instantiated with the provided options
+   */
+  supergraphManager?: SupergraphSchemaManager | never;
+  /**
+   * The logger to be used by this plugin
+   * Defaults to the `console` object
+   */
+  logger?: {
+    info: (message: string) => void;
+    error: (message: string) => void;
+    warn: (message: string) => void;
+  };
+  /**
+   * Allow to customize how a schema loading failure is handled.
+   * A failure happens when the manager failed to load the schema more than the provided max retries
+   * count.
+   * By default, an error is logged and the polling is restarted.
+   * @param error The error encountered during the last fetch tentative
+   * @param delayInSeconds The delay in seconds indicated by GraphOS before a new try
+   */
+  onFailure?: (error: FetchError | unknown, delayInSeconds: number) => void;
 };
 
 export function useManagedFederation(options: ManagedFederationPluginOptions = {}): Plugin {
-  const supergraphManager = new SupergraphSchemaManager(options);
+  const {
+    logger = console,
```

Comment:
You can start the manager within onYogaInit maybe?🤔

Reply:
I've made the change, I think it's better to avoid losing any logs (if an external logger is provided in place of the console) than saving potentially 2ms on a 100ms call :-P